client/src/app/selected-options/selected-options.component.ts

```diff
@@ -0,0 +1,168 @@
+  marker: string;
+  qrCode: string;
+}
+
+export interface GeneratedModel {
+  generated_url: string;
+}
+
+@Component({
+  selector: 'app-dialog-preview',
+  template: `
+    <div>
+      <h1 mat-dialog-title>
+        Please scan the qrCode nad point with your mobile to the image
+      </h1>
+    </div>
+    <div mat-dialog-content>
+      <div fxFlex="calc2Cols">
+        <div>
+          <qr-code [value]="data.qrCode" [size]="250"></qr-code>
+        </div>
+        <div>
+          Or Navigate to:
+        </div>
+        <div>
+          <h2>{{ data.url }}</h2>
+        </div>
+      </div>
+      <div fxFlex="calc4Cols">
+        <img src="{{ data.marker }}" />
+      </div>
+    </div>
+  `
+})
+export class PreviewDialogComponent {
+  constructor(@Inject(MAT_DIALOG_DATA) public data: DialogData) {}
+}
+
+@Component({
+  selector: 'app-selected-options',
+  templateUrl: './selected-options.component.html',
+  styleUrls: ['./selected-options.component.css']
+})
+export class SelectedOptionsComponent implements OnInit {
+  model: Model3d;
+  material: Material;
+  audio;
+  previewDisabled = true;
+  modelComplete = false;
+  materialComplete = false;
+  audioComplete = false;
+
+  @ViewChild('stepper') stepper: MatStepper;
+
+  constructor(
+    private ModelS: Models3dService,
+    private MaterialS: MaterialService,
+    private router: Router,
+    private route: ActivatedRoute,
+    public dialog: MatDialog
+  ) {}
+
+  getModel(id) {
+    this.ModelS.getModel(id).subscribe((mode: Model3d) => {
+      this.model = mode;
+    });
+  }
+
+  getMaterial(id) {
+    this.MaterialS.getMaterial(id).subscribe(
+      (mat: Material) => (this.material = mat)
+    );
+  }
+
+  cb() {}
+
+  selectModel() {
+    this.router.navigate(['/models'], {
+      queryParamsHandling: 'merge'
+    });
+  }
+
+  selectMaterial() {
+    this.router.navigate(['/materials'], {
+      queryParamsHandling: 'merge'
+    });
+  }
+  reset() {
+    this.router.navigate([], {
+      queryParams: {
+        'object-id': null,
+        'material-id': null
+      },
+      queryParamsHandling: 'merge'
+    });
+    this.model = null;
+    this.material = null;
+    this.materialComplete = this.modelComplete = false;
+  }
+
+  generatePreview() {
+    this.ModelS.generatePreview(this.model.id).subscribe(
+      (data: GeneratedModel) => {
+        this.dialog.open(PreviewDialogComponent, {
+          width: '80%',
+          data: {
+            qrCode: data.generated_url,
+            url: data.generated_url,
+            marker:
+              'https://s3.eu-west-3.amazonaws.com/sendspecial/Hiro_marker_ARjs.png'
+          }
+        });
+      }
+    );
+  }
+
+  changeMaterial() {
+    this.router.navigate([], {
+      queryParams: {
+        'material-id': null
+      }
+    });
+    this.material = null;
+  }
+
+  changeModel() {
+    this.router.navigate([], {
+      queryParams: {
+        'object-id': null
+      }
+    });
+    this.model = null;
+  }
+
```

Comment:
Maybe refactor somehow the two methods above into one with ifs? Just for refactoring sake, maybe I'm wrong